### PR TITLE
fix: imported string-key loses annotations

### PR DESCRIPTION
**Pull request by grant-d**
In #744, we added support for string-key annotations in mapped types:

```typescript
/** Currency */
type CurrencyISO = string;
/** Exchange rate */
export type ExchangeRate = Record<CurrencyISO, number>;
```
...which outputs:
```json
{
    "definitions": {
        "ExchangeRate": {
            "type": "object",
            "additionalProperties": {
                "type": "number"
            },
            "propertyNames": {    // <-------- Take special note of this property
                "description": "Currency"
            },
            "description": "Exchange rate"
        }
    }
}
```

However, if the _exact-same_ string type is `imported` instead of being declared _locally_, some of the annotations are lost:
```typescript
// currency.ts
  /** Currency */
  export type CurrencyISO = string;

// main.ts:
  import { Currency } from "./currency.ts"; // <-------- This is the critical change
  
  /** Exchange rate */
  export type ExchangeRate = Record<CurrencyISO, number>;
```
...which outputs:
```json
```json
{
    "definitions": {
        "CurrencyISO" {
        "description": "Currency",
            "pattern": "^[A-Z]{3,3}$",
            "type": "string",
        },
        "ExchangeRate": {
            "type": "object",
            "additionalProperties": {
                "type": "number"
            },
            // <----------- propertyNames missing
            "description": "Exchange rate"
        }
    }
}
```

This PR fixes this issue. The missing `propertyNames` are restored.

# Version

Published prerelease version: `v1.2.1-next.0`

<details>
  <summary>Changelog</summary>

  :tada: This release contains work from a new contributor! :tada:
  
  Thank you, Grant Dickinson ([@grant-d](https://github.com/grant-d)), for all your work!
  
  #### 🐛 Bug Fix
  
  - fix: imported string-key loses annotations [#1293](https://github.com/vega/ts-json-schema-generator/pull/1293) ([@grant-d](https://github.com/grant-d) p-spacek@email.cz)
  
  #### 🔩 Dependency Updates
  
  - chore(deps): bump commander from 9.5.0 to 10.0.0 [#1538](https://github.com/vega/ts-json-schema-generator/pull/1538) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.48.0 to 5.48.1 [#1534](https://github.com/vega/ts-json-schema-generator/pull/1534) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump prettier from 2.8.2 to 2.8.3 [#1533](https://github.com/vega/ts-json-schema-generator/pull/1533) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint from 8.31.0 to 8.32.0 [#1535](https://github.com/vega/ts-json-schema-generator/pull/1535) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.48.0 to 5.48.1 [#1536](https://github.com/vega/ts-json-schema-generator/pull/1536) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump glob from 8.0.3 to 8.1.0 [#1537](https://github.com/vega/ts-json-schema-generator/pull/1537) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.47.1 to 5.48.0 [#1524](https://github.com/vega/ts-json-schema-generator/pull/1524) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint-config-prettier from 8.5.0 to 8.6.0 [#1525](https://github.com/vega/ts-json-schema-generator/pull/1525) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.47.1 to 5.48.0 [#1526](https://github.com/vega/ts-json-schema-generator/pull/1526) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump ajv from 8.11.2 to 8.12.0 [#1527](https://github.com/vega/ts-json-schema-generator/pull/1527) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump commander from 9.4.1 to 9.5.0 [#1528](https://github.com/vega/ts-json-schema-generator/pull/1528) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump prettier from 2.8.1 to 2.8.2 [#1529](https://github.com/vega/ts-json-schema-generator/pull/1529) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @babel/core from 7.20.7 to 7.20.12 [#1530](https://github.com/vega/ts-json-schema-generator/pull/1530) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.47.0 to 5.47.1 [#1520](https://github.com/vega/ts-json-schema-generator/pull/1520) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump safe-stable-stringify from 2.4.1 to 2.4.2 [#1517](https://github.com/vega/ts-json-schema-generator/pull/1517) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/jest from 29.2.4 to 29.2.5 [#1518](https://github.com/vega/ts-json-schema-generator/pull/1518) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps): bump json5 from 2.2.2 to 2.2.3 [#1519](https://github.com/vega/ts-json-schema-generator/pull/1519) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @types/node from 18.11.17 to 18.11.18 [#1521](https://github.com/vega/ts-json-schema-generator/pull/1521) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.47.0 to 5.47.1 [#1522](https://github.com/vega/ts-json-schema-generator/pull/1522) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump eslint from 8.30.0 to 8.31.0 [#1523](https://github.com/vega/ts-json-schema-generator/pull/1523) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/eslint-plugin from 5.46.1 to 5.47.0 [#1514](https://github.com/vega/ts-json-schema-generator/pull/1514) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @typescript-eslint/parser from 5.46.1 to 5.47.0 [#1515](https://github.com/vega/ts-json-schema-generator/pull/1515) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  - chore(deps-dev): bump @babel/core from 7.20.5 to 7.20.7 [#1516](https://github.com/vega/ts-json-schema-generator/pull/1516) ([@dependabot[bot]](https://github.com/dependabot[bot]))
  
  #### Authors: 3
  
  - [@dependabot[bot]](https://github.com/dependabot[bot])
  - Grant Dickinson ([@grant-d](https://github.com/grant-d))
  - Petr Spacek (p-spacek@email.cz)
</details>

### CODE REVIEW

File: src/NodeParser/MappedTypeNodeParser.ts [hunk 1]
```diff
@@ -47,8 +48,18 @@ export class MappedTypeNodeParser implements SubNodeParser {
...
+                let annotations;
+
+                if (constraintType instanceof AnnotatedType) {
+                    annotations = constraintType.getAnnotations();
+                } else if (constraintType instanceof DefinitionType) {
+                    const type = constraintType.getType();
+                    if (type instanceof AnnotatedType) {
+                        annotations = type.getAnnotations();
+                    }
+                }
```
Comment (grant-d) [inline src/NodeParser/MappedTypeNodeParser.ts:61]: Here is the critical change. I am not familiar enough with the object model so perhaps I approached this wrong. This does give the correct result in this case, but perhaps you would do it differently?

File: test/valid-data/type-mapped-annotated-string/main.ts [hunk 2]
```diff
@@ -8,3 +10,8 @@ type CurrencyISO = string;
  * Exchange rate
  */
 export type ExchangeRate = Record<CurrencyISO, number>;
+
+/**
+ * Size chart
+ */
+export type SizeChart = Record<Size, number>;
```
Comment (grant-d) [inline test/valid-data/type-mapped-annotated-string/main.ts:17]: I left the original 'not-imported' `CurrencyISO` type as-is, and added a new `imported` `SizeChart` type to test the issue

File: test/valid-data/type-mapped-annotated-string/schema.json [hunk 3]
```diff
@@ -12,6 +11,22 @@
...
+    },
+    "SizeChart": {
+      "additionalProperties": {
+        "type": "number"
+      },
+      "description": "Size chart",
+      "propertyNames": {
+        "description": "Size",
+        "pattern": "^S|M|L$"
+      },
```
Comment (grant-d) [inline test/valid-data/type-mapped-annotated-string/schema.json:28]: This is what was missing without the fix

File: src/NodeParser/MappedTypeNodeParser.ts (hunk 1 above)
Comment (grant-d) [inline src/NodeParser/MappedTypeNodeParser.ts]: Here is the critical change. I am not familiar enough with the object model so perhaps I approached this wrong. This does give the correct result in this case, but perhaps you would do it differently?

**Comment by domoritz** [comment]
Thanks for the pull request. Can you merge next and make sure that the automated tests run?

**Pull request merged**